File: chroma-manager/chroma_ui_new/test/selenium/util/command-monitor.js
'use strict';

var format = require('util').format;
var isRemoved = require('./is-removed');

var commandMonitor = {
  COMMAND_IN_PROGRESS: '.command-in-progress',
  get commandInProgress () {
    return $(commandMonitor.COMMAND_IN_PROGRESS);
  },
  /**
   * Waits for the spinny to stop spinning.
   * @param {Number} waitTime
   */
  waitForCommandsToFinish: function waitForCommandsToFinish (waitTime) {

    browser.wait(
      isRemoved(commandMonitor.commandInProgress),
      waitTime,
      format('Timeout after %s seconds waiting for commands to complete', waitTime)
    );

  }
};

module.exports = commandMonitor;

File: chroma-manager/tests/selenium/utils/command_monitor.py
from selenium.webdriver.support.wait import WebDriverWait


def wait_for_commands_to_finish(driver, wait):
    return WebDriverWait(driver, wait).until_not(
        lambda driver: driver.find_element_by_css_selector('.command-in-progress'),
        'Timeout after %s seconds waiting for commands to complete' % wait)
